drivers/awinic/aw20216.c

```diff
@@ -55,128 +53,89 @@
-
-#ifndef DRIVER_1_EN
-#    define DRIVER_1_EN C13
-#endif
-
 #ifndef AW_SPI_DIVISOR
 #    define AW_SPI_DIVISOR 4
 #endif
 
-uint8_t g_spi_transfer_buffer[3] = {0};
 uint8_t g_pwm_buffer[DRIVER_COUNT][AW_PWM_REGISTER_COUNT];
 bool    g_pwm_buffer_update_required[DRIVER_COUNT] = {false};
 
-bool AW20216_write_register(pin_t slave_pin, uint8_t page, uint8_t reg, uint8_t data) {
-    // Do we need to call spi_stop() if this fails?
-    if (!AW_SPI_START(slave_pin)) {
+bool AW20216_write(pin_t cs_pin, uint8_t page, uint8_t reg, uint8_t* data, uint8_t len) {
+    static uint8_t s_spi_transfer_buffer[2] = {0};
+
+    if (!spi_start(cs_pin, false, 0, AW_SPI_DIVISOR)) {
+        spi_stop();
         return false;
     }
 
-    g_spi_transfer_buffer[0] = (AWINIC_ID | page | AW_WRITE);
-    g_spi_transfer_buffer[1] = reg;
-    g_spi_transfer_buffer[2] = data;
+    s_spi_transfer_buffer[0] = (AWINIC_ID | page | AW_WRITE);
+    s_spi_transfer_buffer[1] = reg;
 
-    if (spi_transmit(g_spi_transfer_buffer, 3) != SPI_STATUS_SUCCESS) {
+    if (spi_transmit(s_spi_transfer_buffer, 2) != SPI_STATUS_SUCCESS) {
         spi_stop();
         return false;
     }
+
+    if (spi_transmit(data, len) != SPI_STATUS_SUCCESS) {
+        spi_stop();
+        return false;
+    }
+
     spi_stop();
     return true;
 }
 
-bool AW20216_init_scaling(void) {
+static inline bool AW20216_write_register(pin_t cs_pin, uint8_t page, uint8_t reg, uint8_t value) {
+    // Little wrapper so callers need not care about sending a buffer
+    return AW20216_write(cs_pin, page, reg, &value, 1);
+}
+
+static void AW20216_init_scaling(pin_t cs_pin) {
     // Set constant current to the max, control brightness with PWM
-    aw_led led;
-    for (uint8_t i = 0; i < DRIVER_LED_TOTAL; i++) {
-        led = g_aw_leds[i];
-        if (led.driver == 0) {
-            AW20216_write_register(DRIVER_1_CS, AW_PAGE_SCALING, led.r, AW_SCALING_MAX);
-            AW20216_write_register(DRIVER_1_CS, AW_PAGE_SCALING, led.g, AW_SCALING_MAX);
-            AW20216_write_register(DRIVER_1_CS, AW_PAGE_SCALING, led.b, AW_SCALING_MAX);
-        }
-#ifdef DRIVER_2_CS
-        else if (led.driver == 1) {
-            AW20216_write_register(DRIVER_2_CS, AW_PAGE_SCALING, led.r, AW_SCALING_MAX);
-            AW20216_write_register(DRIVER_2_CS, AW_PAGE_SCALING, led.g, AW_SCALING_MAX);
-            AW20216_write_register(DRIVER_2_CS, AW_PAGE_SCALING, led.b, AW_SCALING_MAX);
-        }
-#endif
+    for (uint8_t i = 0; i < AW_PWM_REGISTER_COUNT; i++) {
+        AW20216_write_register(cs_pin, AW_PAGE_SCALING, i, AW_SCALING_MAX);
     }
-    return true;
 }
 
-bool AW20216_soft_enable(void) {
-    AW20216_write_register(DRIVER_1_CS, AW_PAGE_FUNCTION, AW_REG_CONFIGURATION, AW_CONFIG_DEFAULT | AW_CHIPEN);
-#ifdef DRIVER_2_CS
-    AW20216_write_register(DRIVER_2_CS, AW_PAGE_FUNCTION, AW_REG_CONFIGURATION, AW_CONFIG_DEFAULT | AW_CHIPEN);
-#endif
-    return true;
+static inline void AW20216_init_current_limits(pin_t cs_pin) {
```

Comment:
Nitpicking, but this is a single global limit, probably should be called `AW20216_init_current_limit`